test/functional/test_framework/wallet.py

```diff
@@ -55,8 +55,8 @@ def get_utxo(self, *, txid=''):
             index = self._utxos.index(utxo)
         return self._utxos.pop(index)
 
-    def send_self_transfer(self, *, fee_rate=Decimal("0.003"), from_node, utxo_to_spend=None):
-        """Create and send a tx with the specified fee_rate. Fee may be exact or at most one satoshi higher than needed."""
+    def send_self_transfer(self, *, fee_rate=Decimal("0.003"), from_node, utxo_to_spend=None,  submit_tx=True):
```

Comment:
I'm just wondering whether `rpc_generateblock.py` requires a `flat_fee` or a calculated fee? Because, if i'm not mistaken you used `flat_fee` in previous commits

Reply:
The fee was irrelevant to `rpc_generateblock.py` so I used a flat fee to reduce the amount of code in the helper function `create_miniwallet_rawtx`.